core/src/main/java/com/datastax/oss/driver/api/core/type/codec/MappingCodec.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.type.codec;
+
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import com.datastax.oss.driver.api.core.type.DataType;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+/**
+ * A {@link TypeCodec} that maps instances of {@code InnerT}, a supported Java type, to instances of
+ * a target {@code OuterT} Java type.
+ *
+ * <p>This codec can be used to provide support for Java types that are not natively handled by the
+ * driver, as long as there is a conversion path to and from another supported Java type.
+ *
+ * <p>See the <a
+ * href="https://docs.datastax.com/en/developer/java-driver/latest/manual/core/custom_codecs/">driver
+ * documentation on custom codecs</a> for more information and a few usage examples.
+ *
+ * @param <InnerT> The "inner" Java type; must be a driver supported Java type (that is, there must
+ *     exist a codec registered for it).
+ * @param <OuterT> The "outer", or target Java type; this codec will handle the mapping to and from
+ *     {@code InnerT} and {@code OuterT}.
+ */
+public abstract class MappingCodec<InnerT, OuterT> implements TypeCodec<OuterT> {
```

Comment:
maybe instead of `InnerT` and `OuterT` we could simplify it a bit?
so `InnerT` would be `InT` and `OUter` would be `OutT`

Reply:
Shorter is not necessarily better, but is often less clear. Here "InT" and "OutT" could be understood as : "IncomingT" and "OutgoingT" which is not exactly right.